app/views/contexts/calculated_summary_context.py

```diff
@@ -30,65 +35,85 @@ def build_groups_for_section(self, section, return_to_block_id):
             groups or [],
             current_location=current_location,
-            calculation_operator=ValueSourceResolver.get_calculation_operator(
+            calculation=ValueSourceResolver.get_calculation_operator(
                 calculation["calculation_type"]
-            ),
+            )
+            if calculation.get("answers_to_calculate")
+            else calculation["operation"],
         )
 
-        context = {
+        collapsible = block.get("collapsible") or False
+        block_title = block["title"]
+
+        return {
             "summary": {
                 "groups": groups,
                 "answers_are_editable": True,
                 "calculated_question": self._get_calculated_question(
                     calculation, formatted_total
                 ),
-                "title": block.get("title") % dict(total=formatted_total),
-                "collapsible": block.get("collapsible", False),
+                "title": block_title % dict(total=formatted_total),
+                "collapsible": collapsible,
                 "summary_type": "CalculatedSummary",
             }
         }
 
-        return context
-
-    def _build_calculated_summary_section(self, rendered_block, current_location):
+    def _build_calculated_summary_section(
+        self, rendered_block: ImmutableDict, current_location: Location
+    ) -> Mapping:
         """Build up the list of blocks only including blocks / questions / answers which are relevant to the summary"""
-        section_id = self._schema.get_section_id_for_block_id(current_location.block_id)
-        group = self._schema.get_group_for_block_id(current_location.block_id)
+        # type ignores added as block will exist at this point
+        block_id: str = current_location.block_id  # type: ignore
+        group: ImmutableDict = self._schema.get_group_for_block_id(block_id)  # type: ignore
+
+        section_id = self._schema.get_section_id_for_block_id(block_id)
+
         blocks = []
-        answers_to_calculate = rendered_block["calculation"]["answers_to_calculate"]
+        if rendered_block["calculation"].get("answers_to_calculate"):
+            answers_to_calculate = rendered_block["calculation"]["answers_to_calculate"]
+        else:
+            answers_to_calculate = self._schema.get_calculated_summary_answer_ids(
+                rendered_block
+            )
+
         blocks_to_calculate = [
             self._schema.get_block_for_answer_id(answer_id)
             for answer_id in answers_to_calculate
         ]
+
         unique_blocks = list(
-            {block["id"]: block for block in blocks_to_calculate}.values()
+            {block["id"]: block for block in blocks_to_calculate if block}.values()
         )
 
         for block in unique_blocks:
-            if QuestionnaireSchema.is_question_block_type(block["type"]):
+            if block and QuestionnaireSchema.is_question_block_type(block["type"]):
```

Comment:
Same, here why is the `if block` needed.

Reply:
As above 👍